src/operators.jl

```diff
@@ -0,0 +1,55 @@
+### AbstractSciMLOperator Interface
+
+#=
+1. Function call and multiplication: L(du, u, p, t) for inplace and du = L(u, p, t) for
+   out-of-place, meaning L*u and mul!(du, L, u).
+2. If the operator is not a constant, update it with (u,p,t). A mutating form, i.e.
+   update_coefficients!(A,u,p,t) that changes the internal coefficients, and a
+   out-of-place form B = update_coefficients(A,u,p,t).
+3. isconstant(A) trait for whether the operator is constant or not.
+4. islinear(A) trait for whether the operator is linear or not.
+=#
+
+Base.eltype(L::AbstractSciMLOperator{T}) where T = T
```

Comment:
```suggestion
Base.eltype(::Type{AbstractSciMLOperator{T}}) where T = T
```

Reply:
`eltype(zeros(2))`

Reply:
Sure, you can call `eltype` for instances since the generic fallback is `eltype(x) = eltype(typeof(x))`. "However the form that accepts a type argument should be defined for new types." (https://docs.julialang.org/en/v1/base/collections/#Base.eltype).

Reply:
Let's get an issue open on this one. I don't know if breaking this at the same time is a good idea. We'll need to make sure it's not breaking to DiffEqOperators.jl

Reply:
This is not breaking though. It's a better definition.
```julia
julia> struct A end; Base.eltype(::A) = Float64

julia> struct B end; Base.eltype(::Type{B}) = Float64

julia> eltype(A())
Float64

julia> eltype(B())
Float64

julia> eltype(A)
Any

julia> eltype(B)
Float64
```

Reply:
Had to revert because this caused a downstream failure in DiffEqOperators.jl. Can't say I know why, but this isn't the right time to look at that.